### PR TITLE
Auth designs

### PR DESCRIPTION
**Pull request by sam-baraka**
This pull request features the following designs implementation. 
- The splash screen
- The sign in screen
- The sign up screen
- The theme switcher dialod

This is as shown in the screenshots below

Screenshot_1663150039
Screenshot_1663150035
Screenshot_1663150022
Screenshot_1663150018
Screenshot_1663150009
Screenshot_1663150006

### CODE REVIEW

**Review by KendiJ — APPROVED** [review]
Seen

**Review by mulieriq — CHANGES_REQUESTED** [review]
Fix conflicts

**Comment by ndungudedan** [comment]
Can this be merged? 
It does have common things already setup.

File: lib/router/router.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    ),
+    GoRoute(
+      path: '/signin',
+      name: SignInScreen.routeName,
+      builder: (BuildContext context, GoRouterState state) =>
+          const SignInScreen(),
+    ),
+  ],
+  observers: [
+    // FirebaseAnalyticsObserver(analytics: FirebaseAnalytics.instance),
```
Comment (mulieriq) [inline lib/router/router.dart:36]: Removed if not used

File: lib/user_interfaces/splash/splash_screen.dart [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  void initState() {
+    super.initState();
+
+    Future.delayed(const Duration(seconds: 2)).then((value) async {
+      await showDialog(
+          context: context,
+          builder: (context) {
+            return const ThemeDialog();
+          });
+      // ignore: use_build_context_synchronously
```
Comment (mulieriq) [inline lib/user_interfaces/splash/splash_screen.dart:25]: Removed if not used

File: pubspec.yaml [hunk 3]
```diff
@@ -1,28 +1,37 @@
 name: droidcon_app
-description: Flutter app for Kenya's 3rd Android Developer Conference - droidcon to be held in Nairobi from November 16-18th 2022.
-publish_to: 'none'
+description: Flutter app for Kenya's 3rd Android Developer Conference - droidcon
```
Comment (mulieriq) [inline pubspec.yaml:2]: why the space though

**Review by mulieriq — CHANGES_REQUESTED** [review]

**Review by KendiJ — APPROVED** [review]

**Review by mulieriq — APPROVED** [review]

**Pull request merged**